go/vt/sqlparser/ast.go

```diff
@@ -1986,7 +1986,7 @@ type (
 	// supported functions are documented in the grammar
 	CurTimeFuncExpr struct {
 		Name ColIdent
-		Fsp  Expr // fractional seconds precision, integer from 0 to 6
+		Fsp  *Literal // fractional seconds precision, integer from 0 to 6
```

Comment:
I think this should be an int instead. no need to wrap it in a Literal

Reply:
Regarding having a pointer for fsp, the fsp also accepts nil and otherwise an IntLiteral thus we had a pointer Literal and not and int

Reply:
The nil is required because, sometimes there may be no arguments to the function